components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/dao/SQLQueries.java

```diff
@@ -42,15 +45,28 @@ private SQLQueries(){
 
         public static final String GET_APPS_OF_USER_WITH_TENANTAWARE_OR_TENANTUNAWARE_USERNAME = "SELECT " +
                 "CONSUMER_KEY, CONSUMER_SECRET, APP_NAME, OAUTH_VERSION, CALLBACK_URL, GRANT_TYPES, ID, USERNAME, TENANT_ID, " +
+                "USER_DOMAIN FROM IDN_OAUTH_CONSUMER_APPS WHERE (USERNAME=? OR USERNAME=?) AND TENANT_ID=?";
```

Comment:
Yes, we do that when we save an oauth application in `org.wso2.carbon.identity.oauth.da.OAuthAppDAO`

`prepStmt.setString(3, consumerAppDO.getUser().getUserName());
                    prepStmt.setInt(4, IdentityTenantUtil.getTenantId(consumerAppDO.getUser().getTenantDomain()));`